jetstream_pt/layers.py

```diff
@@ -171,14 +167,15 @@ def __call__(self, xq, xk, xv, mask, cache):
       # output = torch.einsum(
       #    "ikjm,ikml->ikjl", scores, values
       # )  # (bs, n_local_heads, seqlen, head_dim)
-      output = torch_xla2.extra.call_jax(
-          jnp.einsum, "ikjm,ikml->ikjl", scores, values
-      )
+      output = torch.einsum("ikjm,ikml->ikjl", scores, values)
       if seqlen == 1:
         output = output[:, :, 0:1, :]
       # For XLA matmul performance boost
       # output = torch.matmul(scores, values)
-      self.env.apply_sharding(output, axis=1)
+      if self.env.shard_on_batch:
+        self.env.apply_sharding(output, axis=0)
+      else:
+        self.env.apply_sharding(output, axis=1)
```

Comment:
Nit, can we use self.env.shard_on_batch to select the axis instead of duplicate the `apply_sharding`?

Reply:
done.